front/src/AuthenticationScreen/index.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from "react";
+import NotificationWindow from "../NotificationWindow/";
+import NotificationTypes from "../NotificationWindow/notificationTypes.js";
+/**
+ * An authentication screen. Upon a submit it will send the token in the
+ * inputfield along with a function to handle the response message.
+ * Authentication screen is responsible for its unrendering.
+ * Author: S.A. Tanja
+ *
+ * Props:
+ *  - handleSubmit: A function that will send the token to the SRRS
+ */
+class AuthenticationScreen extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      token: "",
+      showNotification: false,
+      loggedIn: false
+    };
+  }
+
+  onChange(event) {
+    this.setState({ token: event.target.value });
+  }
+
+  /**
+   * Callback function. In case of a incorrect token, update state to display
+   * error. In case of a correct token, update state to render null.
+   * @param isCorrectToken whether the token sent is the correct token to log in
+   */
+  onResponse(isCorrectToken) {
+    if (!isCorrectToken) {
+      this.setState({ showNotification: true });
+    } else {
+      this.setState({ showNotification: false, loggedIn: true });
+    }
+  }
+
+  /**
+   * Gets the user inputed token, and calls the onSubmit function given from
+   * props, and gives a callback to update state.
+   */
+  onSubmit(event) {
+    const token = this.state.token;
+    this.props.onSubmit(token, isCorrectToken =>
+      this.onResponse(isCorrectToken)
+    );
+    // Suppress refresh of the application
+    event.preventDefault();
+  }
+
+  /**
+   * Update state to unrender the "Incorrect token" notification.
+   */
+  onDismiss() {
+    this.setState({ showNotification: false });
+  }
+
+  render() {
+    const notification = {
+      message: "Incorrect token.",
+      notificationType: NotificationTypes.ERROR
+    };
+    return (
+      <div>
+        {this.state.loggedIn ? null : (
+          <form onSubmit={event => this.onSubmit(event)}>
```

Comment:
Do you even need a form? When not just a field an a button, then you can leave out some form specific things